tests/commands/test-kob-install.sh

```diff
@@ -1 +1,319 @@
+#!/bin/bash
+
+path_to_kob_envs="${KOBMAN_DIR}/envs"
+path_to_kob_env_tests=~/KOBman/tests/envs
+kobman_env_name=${1:-dummyenv}
+version=$2
+KOBMAN_NAMESPACE=${KOBMAN_NAMESPACE:-hyperledgerkochi}
+status="true"
+
+function __test_kob_run
+{
+  __test_kob_init
+  
+  __test_kob_execute
+  __test_kob_validate || return 1
+
+   __test_kob_cleanup 
+
+  __test_kob_install_output
+  
+  
+}
+
+function __test_kob_init {
+  echo "checking for kob..."
+
+  if [[ -d ${KOBMAN_DIR} ]]; then
+
+    echo "kob found"
+    source "${KOBMAN_DIR}/bin/kobman-init.sh"
+  
+  else
+  
+    echo "kob not found"
+    echo "installing kobman...."
+    curl -L https://raw.githubusercontent.com/${KOBMAN_NAMESPACE}/KOBman/master/get.kobman.io | bash > /dev/null 2>&1
+    source "${KOBMAN_DIR}/bin/kobman-init.sh"
+  
+  fi
+  
+  echo "creating and sourcing dummyenv files..."
+  touch $KOBMAN_DIR/var/kobman_env_$kobman_env_name.proc
+  
+  __test_kob_insert_dummyenv_version
+  
+  __test_kob_install_dummyenv_script > $path_to_kob_envs/kobman-$kobman_env_name.sh
+  
+  touch $path_to_kob_env_tests/test-kobman-${kobman_env_name}.sh
+  __test_kob_dummyenv_validation > $path_to_kob_env_tests/test-kobman-${kobman_env_name}.sh
+  
+  source $path_to_kob_envs/kobman-$kobman_env_name.sh
+
+
+}
+
+function __test_kob_execute {
+  echo "executing install command..."
+  kob install --environment ${kobman_env_name} --version $version > ~/output.txt
+#   echo "output.txt"
+#   cat ~/output.txt
+   
+  cat ~/output.txt | grep -q "dummyenv installed"
+  
+  if [[ "$?" == "0" ]]; then
+  
+    echo "0" > $KOBMAN_DIR/var/kobman_env_$kobman_env_name.proc
+  
+  else
+  
+    echo "1" > $KOBMAN_DIR/var/kobman_env_$kobman_env_name.proc    
+  
+  fi
+}
+
+
+function __test_kob_validate {
+  
+  echo "validating install command...."
+  . $path_to_kob_env_tests/test-kobman-${kobman_env_name}.sh  > ~/tmp.txt
+  
+
+  
+  if [[ $(cat $KOBMAN_DIR/var/kobman_env_$kobman_env_name.proc) == "1" ]]; then
+    
+    echo "install command did not execute properly"
+    status="fail"
+    __test_kob_cleanup
+    __test_kob_install_output
+    return 1
+  
+  fi
+
+  tail -1 ~/tmp.txt | grep -q "success"
+  if [[ "$?" != "0" ]]; then
+  
+    echo "output of dummy env validate is: failed"
+    status="fail"
+    __test_kob_cleanup
+    __test_kob_install_output
+    return 1
+  fi
+  
+  if [[ ! -d ~/Dev_${kobman_env_name} ]]; then
+    
+    echo "Dev_${kobman_env_name} was not found in the /home folder"
+    status="fail"
+    __test_kob_cleanup
```

Comment:
validate should only return success or fail.  __test_kob_cleanup should be handled in run.